### DISCUSSION TITLE
How I can create Mirth variables from external script then use them in the destination

### DISCUSSION THREAD

**Discussion by IbrahimAlhaj9** [nextgenhealthcare/connect#5099] [General]
Hi Everyone, I want to create multiple variables inside external JavaScript file instead of create them from transformer(Add step(type: Mapper))
My code:

```
var case;

try {
    case = msg['ORC']['ORC.3']['ORC.3.1'].toString();
} catch (e) {
    case = '';
}

var Name;

try {
    Name = msg['ORC']['ORC.3']['ORC.3.1'].toString();
} catch (e) {
    Name = '';
}

var FileName;
FileName = case + "_" + Name;
FileName = FileName.toUpperCase();
channelMap.put('FileName', validate(FileName, '', new Array()));

```
how i can put it in an external file then use the variable in the destination in order to rename the file?

Thank you

**Comment by pacmano1** [comment]
What do you mean by "external Javascript file"?     I don't think I understand your grammar when you post sometimes (e.g. your question about tildes highlights that),  perhaps provide more screenshots or more detail.

If you are creating channel map variables in mapper steps or anywhere else and want to use that channel map variable as the file name for a file writer, your channel map variable can be drag dropped from the right hand side list of variables to the file name of the file writer.

**Reply by IbrahimAlhaj9** [reply]
@pacmano1 sorry if you don't understand me, this is my fault.

I mean by "external Javascript file" like the following:
![file](https://user-images.githubusercontent.com/54806435/158753589-ae72d647-eacb-4a61-a8f9-e7aef6921d0a.PNG)
inside the test.js I put the following code:
```
function test(msg)
{
var case;

try {
    case = msg['ORC']['ORC.3']['ORC.3.1'].toString();
} catch (e) {
    case = '';
}

var Name;

try {
    Name = msg['ORC']['ORC.3']['ORC.3.1'].toString();
} catch (e) {
    Name = '';
}

var FileName;
FileName = case + "_" + Name;
FileName = FileName.toUpperCase();
channelMap.put('FileName', validate(FileName, '', new Array()));
}
```
My question is how I can use the "FileName" variable in the destination when I am using this method?
Thank you

**Reply by pacmano1** [reply]
Got it and thanks for the screenshot.   I have _never_ used a transformer external javascript step since I have been using mirth.  Curious, why are you using it?   My earlier comment still applies though (second paragraph in drag drop).   Using a javascript step (not the external one) is typical though where you just build stuff natively in JS vs use the mapper steps.

**Reply by pacmano1** [reply]
What is your ultimate goal here?  Name a file based on one or more segments?  What will be the contents of the file?    Something from the message itself?    

Explaining that might result in a fast answer overall.

**Reply by IbrahimAlhaj9** [reply]
@pacmano1 thank you for your explanation.
I usually build a driver(translation message hl7 to hl7) using external script. But now I have a thing that I want to change the destination file name based on the input message data .
So I would like to define the file name variable outside the transformer like the driver code(JavaScript code).
Is this doable in mirth?
Thank you.

**Reply by IbrahimAlhaj9** [reply]
@pacmano1 any update on my last comment?

**Reply by siddharthsharma9258** [reply]
You can always perform the same tasks with JS based transformer steps. I think there is updated information about the external scripts in the  mirth guide. It is rarely used.

**Reply by IbrahimAlhaj9** [reply]
yes @siddharthsharma9258 I tried this method and it was worked but I am searching to do the same but from external script if this is doable
Thank you

**Reply by pacmano1** [reply]
No clue, I don't even see a reason why you would want to do this.  And I know other seasoned mirth folks that don't use that, so I'd suggest you listen to folks that have been using mirth a long time and don't use it either (simply from a support perspective here of course, do what you like anyway).

**Reply by pacmano1** [reply]
Also please go back to your other threads and mark an answer.
